execute/plugin.go

```diff
@@ -227,32 +227,32 @@ func getPendingReportsForExecution(
 		finalizedMsgSet := mapset.NewSet[cciptypes.SeqNum]()
 		allExecutedMsgSet := mapset.NewSet[cciptypes.SeqNum]()
 
-		for _, seqRange := range ranges {
+		if len(ranges) != 0 {
 			// Get all executed messages
-			allMessages, err := ccipReader.ExecutedMessages(ctx, selector, seqRange, primitives.Unconfirmed)
+			allMessages, err := ccipReader.ExecutedMessages(ctx, selector, ranges, primitives.Unconfirmed)
 			if err != nil {
-				return nil, nil, nil, fmt.Errorf("get %d executed messages in range %v: %w", selector, seqRange, err)
+				return nil, nil, nil, fmt.Errorf("get %d executed messages in range %v: %w", selector, ranges, err)
 			}
-			allExecutedMsgSet = allExecutedMsgSet.Union(mapset.NewSet(allMessages...))
+			allExecutedMsgSet = mapset.NewSet(allMessages...)
 
 			// Get finalized messages
-			finalizedMessages, err := ccipReader.ExecutedMessages(ctx, selector, seqRange, primitives.Finalized)
+			finalizedMessages, err := ccipReader.ExecutedMessages(ctx, selector, ranges, primitives.Finalized)
 			if err != nil {
-				return nil, nil, nil, fmt.Errorf("get finalized %d executed messages in range %v: %w", selector, seqRange, err)
+				return nil, nil, nil, fmt.Errorf("get finalized %d executed messages in range %v: %w", selector, ranges, err)
 			}
-			finalizedMsgSet = finalizedMsgSet.Union(mapset.NewSet(finalizedMessages...))
+			finalizedMsgSet = mapset.NewSet(finalizedMessages...)
 		}
```

Comment:
With the suggested API I think this would be:

```golang
executed, err := ccipReader.ExecutedMessages(ctx,map[cciptypes.ChainSelector][]cciptypes.SeqNumRange{selector: ranges}, primitives.Unconfirmed)
// ... rest of logic
```